Target preview support & misc updates

swarna04 opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description

Ref: [mOB-15329]

- Added missing preview functionality, along with functional test updates
- Fixed make archive issues (specify SwiftLint as pod dependency)
- Added support for dSym generation upon make archive

## Related Issue

<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->

## Motivation and Context

<!--- Why is this change required? What problem does it solve? -->

## How Has This Been Tested?

<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## Screenshots (if appropriate):

## Types of changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

sbenedicadb commented on AEPTarget.xcodeproj/project.pbxproj:
```diff
@@ -822,7 +822,7 @@
 			runOnlyForDeploymentPostprocessing = 0;
 			shellPath = /bin/zsh;
-			shellScript = "cd ${PROJECT_DIR}\nif which swiftformat >/dev/null; then\n   swiftformat .\nelse\necho \"error: SwiftFormat not installed, download from https://github.com/nicklockwood/SwiftFormat\"\nfi\n";
+			shellScript = "cd ${PROJECT_DIR}\nif which ./Pods/SwiftLint/swiftlint >/dev/null; then\n  ./Pods/SwiftLint/swiftlint\nelse\n  echo \"error: SwiftLint not installed, download from https://github.com/realm/SwiftLint\"\nfi\n";
```
i guess since `SwiftLint` is a dependency in the podfile now, maybe this error message should prompt the developer to run `pod update` instead of pointing to the github repo?

swarna04 replied:
I am of the same opinion, but I wanted to keep it consistent with the changes we made for Mobile Core https://github.com/adobe/aepsdk-core-ios/pull/722/files. I'll update it anyways.

sbenedicadb commented on AEPTarget/Sources/TargetDeliveryRequestBuilder.swift:
```diff
@@ -54,7 +55,17 @@ enum TargetDeliveryRequestBuilder {
 
-        return TargetDeliveryRequest(id: targetIDs, context: context, experienceCloud: experienceCloud, prefetch: prefetch, execute: execute, notifications: notifications, environmentId: environmentId, property: property)
+        var qaMode: [String: AnyCodable]?
+        if
+            let qaModeParameters = qaModeParameters,
+           !qaModeParameters.isEmpty,
+           let qaModeData = qaModeParameters.data(using: .utf8)
```
this formatting is rough - is it from swift format?  wondering if we need to revisit our rules if so...

swarna04 replied:
Yeah, i've fixed the alignment here but it's essentially [opening_brace](https://realm.github.io/SwiftLint/opening_brace.html) (swiftlint) and [wrapMultilineStatementBraces](https://github.com/nicklockwood/SwiftFormat/blob/master/Rules.md#wrapmultilinestatementbraces) which are at odds here. I've contributed little to the formatting/ linting rules here and I don't think we have a standard (rules) that we follow across the extensions here.
I can essentially disable wrapMultilineStatementBraces in swiftformat config if it's desirable, let me know.

sbenedicadb commented on AEPTarget/Sources/TargetDeliveryRequestBuilder.swift:
```diff
@@ -54,7 +55,17 @@ enum TargetDeliveryRequestBuilder {
         }
 
-        return TargetDeliveryRequest(id: targetIDs, context: context, experienceCloud: experienceCloud, prefetch: prefetch, execute: execute, notifications: notifications, environmentId: environmentId, property: property)
+        var qaMode: [String: AnyCodable]?
+        if
+            let qaModeParameters = qaModeParameters,
+           !qaModeParameters.isEmpty,
+           let qaModeData = qaModeParameters.data(using: .utf8)
+        {
+            let qaModeDict = try? JSONSerialization.jsonObject(with: qaModeData, options: []) as? [String: Any]
+            qaMode = AnyCodable.from(dictionary: qaModeDict?[TargetConstants.TargetJson.QA_MODE] as? [String: Any])
+        }
```
this section is begging to be shared internally in an extension (json/dictionary/string conversions).  maybe something we can bring up with the core team to get into services?

swarna04 replied:
Sure, let's discuss this further in the SDK guild meeting.

sbenedicadb reviewed (approved):
looks good - a couple of comments we can discuss

swarna04 commented:
@sbenedicadb Thank you for your review on this one, I've addressed a couple of items from your feedback here:

- Fixed indentation for multiline if, no updates to formatting rules for now.
- Fixed the error message for SwiftLint when running script in build phases.

Let me know if that's good to merge the pr.
